### PR TITLE
Adds support for registerElement for web components

### PR DESCRIPTION
**Pull request by jamesgpearce**
Just adds the registerElement function as per http://w3c.github.io/webcomponents/spec/custom/#extensions-to-document-interface-to-register

### CODE REVIEW

File: lib/dom.js [hunk 1]
```diff
@@ -202,6 +202,12 @@ declare class HTMLCollection {
     [index: number]: Element;
 }
 
+// from http://w3c.github.io/webcomponents/spec/custom/#extensions-to-document-interface-to-register
+declare class ElementRegistrationOptions {
```
Comment (mroch) [inline lib/dom.js]: does this need to be a class, or can it just be an object type, e.g. `type ElementRegistrationOptions = {...`? it doesn't look like you `new` it

**Comment by jamesgpearce** [comment]
:v:

**Comment by jamesgpearce** [comment]
wait, I messed that up completely

File: lib/dom.js [hunk 2]
```diff
@@ -263,6 +269,8 @@ declare class Document extends Node {
     xmlStandalone: boolean;
     xmlVersion: string;
 
+    registerElement(type: string, options: ElementRegistrationOptions): any;
```
Comment (gabelevi) [inline lib/dom.js]: I believe the spec says that `options` should be optional.

**Comment by jamesgpearce** [comment]
yes indeed, thanks

**Comment by gabelevi** [comment]
Great, thanks for addressing my nit! Merging...

**Pull request closed without merging**